### PR TITLE
Protecting deletion of namespaces

### PR DESCRIPTION
**Pull request by lzecca78**
Hi everyone 👋 , 
as @nandajavarma suggested, I am splitting the PR https://github.com/sighupio/fury-kubernetes-opa/pull/27/commits in 2 because it involves also the introduction of git-chglog that deserves to be in another chapter of the book.
So refer for this to the PR #27

### CODE REVIEW

**Comment by nandajavarma** [comment]
Thanks for making this change @lzecca78 ! LGTM! :shipit:

**Review by nandajavarma — APPROVED** [review]

**Comment by lzecca78** [comment]
Can we merge this? @ralgozino @angelbarrera92 ? Do you think that is everything ok or is still missing anything?

File: docs/releases/unreleased.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+SIGHUP team maintains this module updated and tested. That is the main reason why we worked on this new release.
+With the Kubernetes 1.21 release, it became the perfect time to start testing this module against this Kubernetes
+release.
+
+Continue reading the [Changelog](#changelog) to discover them:
+
+## Changelog
+
+- Add a Constraint Template to protect namespaces for being deleted.
```
Comment (ralgozino) [inline docs/releases/unreleased.md:11]: Can we explain here or (maybe better) in the readme how to use the new rule? The annotations, etc.

File: katalog/gatekeeper/rules/constraints/kustomization.yaml [hunk 2]
```diff
@@ -11,6 +11,7 @@ resources:
   - must_have_unique_ingresses.yml
   - must_have_liveness_probe.yml
   - must_have_readiness_probe.yml
+  - must_have_namespace_label_to_be_safely_deleted.yml
```
Comment (ralgozino) [inline katalog/gatekeeper/rules/constraints/kustomization.yaml:14]: I'm not sure about having this one ON by default. What do the rest think about it?

File: docs/releases/unreleased.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+## Upgrade path
+
+To upgrade this core module from `v1.3.1`, you need to download this new version, then apply the
+`kustomize` project. No further action is required.
+
```
Comment (ralgozino) [inline docs/releases/unreleased.md:17]: We should say that there's a breaking change: now we are watching the DELETE events also. If people are using their own rules they must be aware of this change, they probably might have to do the same fix that we had to.

**Review by ralgozino — CHANGES_REQUESTED** [review]

File: katalog/gatekeeper/rules/constraints/kustomization.yaml (hunk 2 above)
Comment (lzecca78) [reply katalog/gatekeeper/rules/constraints/kustomization.yaml:14]: I think that once is clear how to handle this, is a great added value to have this onboard by default. In the past happened to me to accidentally delete a namespace in a production environment and I whish there was a such feature :)

**Review by nandajavarma — APPROVED** [review]

**Pull request merged**